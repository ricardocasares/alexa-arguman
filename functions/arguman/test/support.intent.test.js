const app = require("../app");
const topic = require("./requests/topic");
const support = require("./requests/support");
const no_contention = require("./requests/support_no_contention");
jest.mock("../lib");

describe("arguman", () => {
  test("should respond to support intents", () => {
    return expect(app.request(support)).resolves.toMatchSnapshot();
  });

  test("should respond to support intents without contention", () => {
    return expect(app.request(no_contention)).resolves.toMatchSnapshot();
  });
});
