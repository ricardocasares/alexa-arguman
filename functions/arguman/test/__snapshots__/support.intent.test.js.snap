// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arguman should respond to support intents 1`] = `
Object {
  "response": Object {
    "directives": Array [],
    "outputSpeech": Object {
      "ssml": "<speak>Support rate for this contention is 100%</speak>",
      "type": "SSML",
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {
    "contention": Object {
      "absolute_url": "http://en.arguman.org/marxism-is-an-inherently-violent-revolutionary-ideology",
      "date_creation": "2018-01-18T20:30:51.550Z",
      "date_modification": "2018-01-19T17:47:51.696Z",
      "id": 4023,
      "premise_count": 2,
      "sender": Object {
        "id": 29043,
        "username": "thundax",
      },
      "support_rate": 100,
      "title": "Marxism is an inherently violent revolutionary ideology",
    },
  },
  "version": "1.0",
}
`;

exports[`arguman should respond to support intents without contention 1`] = `
Object {
  "response": Object {
    "directives": Array [],
    "outputSpeech": Object {
      "ssml": "<speak>Please choose a topic first</speak>",
      "type": "SSML",
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {},
  "version": "1.0",
}
`;
